### ISSUE TITLE
Unable to create Authorizer using Object notation in serverless.yml

### ISSUE THREAD

**Issue by nguyer** [serverless/serverless#1948]
Serverless Framework Version: 1.0.0-beta.2

Operating System: Mac OS X 10.11.4

Expected Behavior: API Gateway Authorizer created

Actual Behavior: API Gateway Authorizer fails to create, with the following message:

```
  Serverless Error ---------------------------------------

     An error occurred while provisioning your cloudformation:
     The following resource(s) failed to update: [authorizeMessagingUserAuthorizer].


  Get Support --------------------------------------------
     Docs:          v1.docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
```

Description: When I attempt to use the Object notation for defining an Authorizer, Serverless fails to create the API Gateway Authorizer. The documented example that I am using is here: https://github.com/serverless/serverless/blame/85f4084e6b0fd4a6d763ace8cd0db82817bbc712/lib/plugins/aws/deploy/compile/events/apiGateway/README.md#L85

My `serverless.yml` file looks like this:

```

service: my-serverless-service

provider:
  name: aws
  runtime: nodejs4.3

functions:
  authorizeMessagingUser:
    handler: functions/authorizeMessagingUser.handler
    memorySize: 256
  sendMessage:
    handler: functions/sendMessage.handler
    memorySize: 256
    events:
      - http:
          path: applications/{applicationId}/messages
          method: post
          authorizer:
            name: authorizeMessagingUser
            identitySource: method.request.header.Authorization
  getMessage:
    handler: functions/getMessage.handler
    memorySize: 256
    events:
      - http:
          path: applications/{applicationId}/messages
          method: get
          authorizer:
            name: authorizeMessagingUser
            identitySource: method.request.header.Authorization
```

If I change one of the events to look like:

```
  sendMessage:
    handler: functions/sendMessage.handler
    memorySize: 256
    events:
      - http:
          path: applications/{applicationId}/messages
          method: post
          authorizer: authorizeMessagingUser
```

Then everything works just fine. It seems like something different is happening behind the scenes when I use the object notation.

**Comment by tomesch** [comment]
I ran into the same problem. The problem arrises when using the object notation without providing a TTL.

When using the object notation without providing a TTL, [`String(authorizer.resultTtlInSeconds)`](https://github.com/serverless/serverless/blob/85f4084e6b0fd4a6d763ace8cd0db82817bbc712/lib/plugins/aws/deploy/compile/events/apiGateway/lib/authorizers.js#L46) returns "undefined", which makes the creation of the authorizer fail.

#1970 should fix this

**Comment by flomotlik** [comment]
Fixed and merged with #1970
